Sources/Event Logging/TracksService.m

```diff
@@ -317,12 +320,24 @@ - (void)startNetworkMonitor {
         return;
     }
 
-    __weak typeof(self) weakSelf = self;
     self.networkMonitor = nw_path_monitor_create();
-    nw_path_monitor_set_update_handler(_networkMonitor, ^(nw_path_t  _Nonnull path) {
+
+    // Create and set a queue for where the monitor can execute it's event handlers.
+    //
+    // Without one, the update handler doesn't get called when the monitor starts.
+    dispatch_queue_attr_t attrs = dispatch_queue_attr_make_with_qos_class(
+                                                                          DISPATCH_QUEUE_SERIAL,
+                                                                          QOS_CLASS_UTILITY,
+                                                                          DISPATCH_QUEUE_PRIORITY_DEFAULT
+                                                                          );
+    self.networkMonitorQueue = dispatch_queue_create("com.automattic.tracks.network.monitor", attrs);
```

Comment:
Apple's [docs](https://developer.apple.com/documentation/dispatch/dispatch_queue) (and [various Apple engineers](https://mjtsai.com/blog/2021/03/16/underused-and-overused-gcd-patterns/)) recommend using `dispatch_queue_create_with_target` instead to create a serial queue that will target one of the existing concurrent views, rather than creating a new root queue. Creating a root queue will usually result in a permanent thread dedicated to this queue, which is definitely overkill for this particular use case.

Bonus: if we use the `target` API, then I don't think we need to create the `attrs` above at all.

I really wish this were documented better, but it's not. It's all in WWDC videos and Twitter lore from Apple engineers.



```suggestion
    dispatch_queue_t utilityQueue = dispatch_get_global_queue(QOS_CLASS_UTILITY, 0);
    self.networkMonitorQueue = dispatch_queue_create_with_target("com.automattic.tracks.network.monitor", DISPATCH_QUEUE_SERIAL, utilityQueue);
```

Reply:
@bjhomer, I tried this approach but it behaved as before: When the demo app launches, the update handler block is not called.

I pushed that version to a dedicated branch if you want to play with it https://github.com/Automattic/Automattic-Tracks-iOS/commit/2ca8b3bf817d1aa437087f40f0c8a36512201961 .

In the meantime, I applied your [other suggestion](https://github.com/Automattic/Automattic-Tracks-iOS/pull/199#discussion_r822667556) and verified it works following the manual script from the PR description.

What do you think of merging this version in, so we can ship the fix in WordPress 19.5, which is scheduled for the coming Monday, and then figure out why the code that Apple recommends doesn't work?

Thanks!

Reply:
@mokagio I just checked out 2ca8b3b. It looks like it's missing a call to `network_monitor_set_queue`, which is probably why it wasn't working. I added in that call and verified that it does hit the breakpoint upon launch.

If you'd rather merge in this version, though, I'm willing to accept that.

Reply:
@bjhomer 🤦‍♂️... I had a feeling I missed something, but couldn't put my finger on it.

I incorporated your suggestion and verified it worked, as you said 🎉 Thanks!

Given my track record in this PR, I dismissed your review and asked for a new one. Much appreciated 🙇‍♂️